File: connection-backend/src/main/java/de/florianschmitt/service/util/CustomELocalizedDTOComparator.java
package de.florianschmitt.service.util;

import java.util.Comparator;

import de.florianschmitt.model.rest.ELocalizedDTO;

public class CustomELocalizedDTOComparator implements Comparator<ELocalizedDTO> {

    public static CustomELocalizedDTOComparator INSTANCE = new CustomELocalizedDTOComparator();

    @Override
    public int compare(ELocalizedDTO left, ELocalizedDTO right) {
        String leftLocale = left.getLocale();
        String rightLocale = right.getLocale();

        Comparator<String> comparator = Comparator.comparingInt(CustomELocalizedDTOComparator::localeOrder);

        return comparator.compare(leftLocale, rightLocale);
    }

    public static int localeOrder(String locale) {
        switch (locale) {
            case "de":
                return 0;
            case "en":
                return 1;
            case "ar":
                return 2;
            default:
                return 3;
        }
    }

}

File: src/main/kotlin/de/florianschmitt/service/util/CustomELocalizedDTOComparator.kt
package de.florianschmitt.service.util

import de.florianschmitt.model.rest.ELocalizedDTO
import java.util.*

class CustomELocalizedDTOComparator : Comparator<ELocalizedDTO> {

    override fun compare(left: ELocalizedDTO, right: ELocalizedDTO): Int {
        val leftLocale = left.locale
        val rightLocale = right.locale

        val comparator = Comparator.comparingInt<String> { localeOrder(it) }
        return comparator.compare(leftLocale, rightLocale)
    }

    companion object {
        val INSTANCE = CustomELocalizedDTOComparator()

        fun localeOrder(locale: String) = when (locale) {
            "de" -> 0
            "en" -> 1
            "ar" -> 2
            else -> 3
        }
    }
}
